### DISCUSSION TITLE
Does it make sense to prune completed event publication entries from the database?

### DISCUSSION THREAD

**Discussion by matiwinnetou** [spring-projects/spring-modulith#447] [Q&A] [answered]
Once an event is completed and stored in JPA/JDBC in event_publication table does it make sense from the Spring Modulith point of view to keep it?

Is there any plan to implement event prunning, e.g. if flag is set then all completed events are regularly removed from database.

My assumption is that those events that are completed are persisted only for potential audit trail purposes, is this correct?

**Comment by odrotbohm** [comment, marked as answer]
Your understanding is generally correct. We currently do not prune them directly as there might be a chance for them to be useful information about what the system did at a certain point of time. They essentially act as proof the listener code executed successfully.

Spring Modulith 1.1 shipped API to deal with completed event publications in the [`CompletedEventPublications`](https://docs.spring.io/spring-modulith/docs/1.1.x/api/org/springframework/modulith/events/CompletedEventPublications.html) and handle them. Convenience methods for pruning ones matching a filter or having been completed for a particular amount of time exist.
